credentials/google/google.go

```diff
@@ -35,63 +35,56 @@ const tokenRequestTimeout = 30 * time.Second
 
 var logger = grpclog.Component("credentials")
 
-// NewDefaultCredentials returns a credentials bundle that is configured to work
-// with google services.
+// DefaultCredentialsOptions constructs options to build DefaultCredentials.
+type DefaultCredentialsOptions struct {
+	// PerRPCCreds is a  per RPC credentials that is passed to a bundle.
+	PerRPCCreds credentials.PerRPCCredentials
+}
+
+// NewDefaultCredentialsWithOptions returns a credentials bundle that is
+// configured to work with google services.
 //
 // This API is experimental.
-func NewDefaultCredentials() credentials.Bundle {
+func NewDefaultCredentialsWithOptions(opts DefaultCredentialsOptions) credentials.Bundle {
+	perRPC := opts.PerRPCCreds
+	if perRPC == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), tokenRequestTimeout)
+		defer cancel()
+		var err error
+		perRPC, err = oauth.NewApplicationDefault(ctx)
+		if err != nil {
+			logger.Warningf("google default creds: failed to create application oauth: %v", err)
+		}
+	}
```

Comment:
With this minor, probably fine behavior change you should just go ahead and `s/newPerRPCCreds/perRPCCreds/` and not make it a function anymore.  It is now literally equivalent.

Reply:
It's actually different...

perRPC is set only with mode `CredsBundleModeFallback` and `CredsBundleModeBackendFromBalancer`.
And is nil for `CredsBundleModeBalancer`. Because mode balancer is ALTS, and ALTS already has the user identity?

Or I can make another field `perRPCCredsBackup`, and only set `perRPCCreds` for those two modes.

Reply:
Oh, I see.  I didn't realize there already was a `perRPCCreds` field.  How about call this field `perRPCCreds` and rename the other one to `activePerRPCCreds`?

Reply:
> rename the other one to activePerRPCCreds?

Or a `perRPCCredsActive bool`?

Reply:
Made `options` be a field. PTAL (the last commit)